### PR TITLE
Tweaks to frequency spectrum and sound ellipse in visualizer

### PR DESCRIPTION
**Pull request by whooie**
Brings back the linear(-ish) axis scaling from pre-0.9.x (see #453) while keeping DFT adjustments made in #397 and adding config options `visualizer_spectrum_log_scale_x` and `visualizer_spectrum_log_scale_y` for log scaling on either/both axes.

Also switches the character used for the sound ellipse to `Config.visualizer_chars[0]` from `Config.visualizer_chars[1]` (by default this is the round character rather than the bar) because I thought it looked silly otherwise.

Here's a sample of what the linear scaling looks like in my terminal:

and with `visualizer_smooth_look = yes`:

A couple additional notes:
- I noticed that the cubic interpolation used with log-scaling caused a problem when linear-scaling where the leftmost column would not be cropped properly and instead wrap onto the top/bottom edges of the drawing area:
  
  This usually isn't a problem with log scaling because the DFT always goes to zero at those low frequencies (assuming small enough `visualizer_spectrum_hz_min`), but here it's quite bothersome, so when doing x-axis linear scaling, I set the height for bins not directly computed by the DFT to zero rather than interpolate, since I think it's not needed in this case anyway.
- `visualizer_spectrum_gain` is still in dB, for consistency with the `visualizer_spectrum_log_scale_y = yes` case. The constants setting the vertical scale are set to my personal preference for default `gain = 10` and are slightly adjusted compared to pre-0.9.x (slightly less high-frequency boost).

### CODE REVIEW

File: doc/config [hunk 1]
```diff
@@ -141,6 +141,11 @@
 #
 #visualizer_spectrum_hz_max = 20000
 #
+## Use log scaling for the frequency spectrum axes
+#
+#visualizer_spectrum_log_scale_x = no
```
Comment (arybczak) [inline doc/config:146]: These should be `yes` by default.

File: src/screens/visualizer.cpp [hunk 2]
```diff
@@ -95,7 +97,7 @@ Visualizer::Visualizer()
 	memset(m_fftw_input, 0, sizeof(double)*DFT_TOTAL_SIZE);
 	m_fftw_output = static_cast<fftw_complex *>(fftw_malloc(sizeof(fftw_complex)*m_fftw_results));
 	m_fftw_plan = fftw_plan_dft_r2c_1d(DFT_TOTAL_SIZE, m_fftw_input, m_fftw_output, FFTW_ESTIMATE);
-	m_dft_logspace.reserve(500);
+    m_dft_freqspace.reserve(500);
```
Comment (arybczak) [inline src/screens/visualizer.cpp:100]: Please make the indentation use tabs, not spaces. This applies to all of code in this PR.

File: src/screens/visualizer.h [hunk 3]
```diff
@@ -111,9 +113,11 @@ struct Visualizer: Screen<NC::Window>, Tabbable
 	const double DYNAMIC_RANGE;
 	const double HZ_MIN;
 	const double HZ_MAX;
+	const bool LOG_SCALE_X;
```
Comment (arybczak) [inline src/screens/visualizer.h:116]: These are not needed, you can use values from `Config` directly.

**Review by arybczak — COMMENTED** [review]
Thanks, that seems to do what it advertises.

However, I can see that the leftmost column is always empty. Why that is?

**Comment by whooie** [comment]
> Thanks, that seems to do quite well what it advertises 
> 
> However, I can see that the leftmost column is always empty. Why that is?

Hm, I guess I had my font size to small to have noticed that. It's probably because that column is not guaranteed to be directly computed by the DFT, and (to prevent that drawing bug I found that occurs when I have cubic interpolation on) I set those columns to zero height. Not sure what the proper fix for that is.

**Comment by arybczak** [comment]
> that column is not guaranteed to be directly computed by the DFT

Can you explain?

**Comment by whooie** [comment]
This is what I think is happening: the DFT can only be computed for a set number of frequencies. When it's plotted on the terminal, you'll eventually run into a case where you have less than one DFT data point being used to draw one column of characters, and in this case you have to do some interpolation to make the plot look continuous. This happens all the time when plotting on a log-scaled horizontal axis, but for some reason happens only for the leftmost column on the linear-scaled one.

**Comment by arybczak** [comment]
I know it's been ages, but if you're still interested in getting this merged, can you please:

1. Rebase the PR.
2. Clean up the diff (there are tons of irrelevant indentation changes, see the "Files changed" tab).

**Comment by whooie** [comment]
Hey, sorry-- yeah this really fell off my radar for a while. Will do when I can!

**Comment by whooie** [comment]
Hi, seems like this should be ready to go. Let me know if I need to do anything else (sorry if anything is wrong, this is my first time contributing to anyone else's project).

**Comment by arybczak** [comment]
Thanks!

**Pull request merged**